src/_P039_Thermosensors.ino

```diff
@@ -506,21 +535,176 @@ boolean Plugin_039(byte function, struct EventStruct *event, String& string)
       }
       else
       {
-        String log;
-        if((log.reserve(66u))) {
-          log = F("P039 :  ");
-          log += getTaskDeviceName(event->TaskIndex);
-          log += F(" : ");
-          log += F("No Sensor attached !");
-          UserVar[event->BaseVarIndex]     = NAN;
-          UserVar[event->BaseVarIndex + 1] = NAN;
-          addLog(LOG_LEVEL_INFO, log);
+        if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+          String log;
+          if((log.reserve(66u))) {
+            log = F("P039 :  ");
+            log += getTaskDeviceName(event->TaskIndex);
+            log += F(" : ");
+            log += F("No Sensor attached !");
+            UserVar[event->BaseVarIndex]     = NAN;
+            UserVar[event->BaseVarIndex + 1] = NAN;
+            addLog(LOG_LEVEL_INFO, log);
+          }
         }
         success = false;
       }
 
       break;
     }
+
+    case PLUGIN_TIMER_IN: 
+    {
+      P039_data_struct *P039_data = static_cast<P039_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      uint8_t CS_pin_no = get_SPI_CS_Pin(event);
+
+      // Get the MAX Type (6675 / 31855 / 31856)
+      byte MaxType = P039_MAX_TYPE;
+
+      switch (MaxType) 
+      {
+        case P039_MAX31865:
+        {
+          if((nullptr != P039_data)){
+            switch(event->Par1)
+              {
+                case MAX31865_BIAS_ON_STATE:
+                {
+                  // calc delta since last call
+                  unsigned long delta = millis() - P039_data->timer;
```

Comment:
Use `timePassedSince()`
Return value of that function is a signed integer for a reason :)

Reply:
On top -  it is only requried for debugging purposes. I will add depedency of DEBUG logging level as well.

Reply:
Solved - check !